tests/integration/conftest.py

```diff
@@ -674,3 +719,109 @@ def kernel(shell):
+    # If there is already an `exclude` section, append to the existing value
+    def _exclude_packages(exclude):
+        for pkg in exclude:
+            if config.has_option("main", "exclude"):
+                pre_existing_value = config.get("main", "exclude")
+                config.set("main", "exclude", f"{pre_existing_value} {pkg}")
+            else:
+                config.set("main", "exclude", pkg)
+
+        with open(yum_config, "w") as configfile:
+            config.write(configfile, space_around_delimiters=False)
+
+        assert config.has_option("main", "exclude")
+        assert all(pkg in config.get("main", "exclude") for pkg in exclude)
+
+    yield _exclude_packages
+
+    # Clean up
+    assert shell(f"mv {backup_dir}/yum.conf {yum_config}").returncode == 0
+    assert shell(f"rm -r {backup_dir}").returncode == 0
+
+
+def _add_client_tools_repo_oracle(shell):
+    """
+    Helper function.
+    Runs only on Oracle Linux system
+    Create an ubi repo for its respective major version to install subscription-manager from.
+    """
+    if "oracle" in SYSTEM_RELEASE_ENV:
+        with open("/etc/system-release", "r") as sysrelease_file:
+            content = sysrelease_file.read()
+            sysver = get_system_version_number(content)
+
+        repo_url = "https://ftp.redhat.com/redhat/client-tools/client-tools-for-rhel-8-x86_64.repo"
+        if sysver.major == 7:
+            repo_url = "https://ftp.redhat.com/redhat/client-tools/client-tools-for-rhel-7-server.repo"
+
+        # Add the redhat-release GPG key
+        assert shell(f"curl -o /etc/yum.repos.d/client-tools-for-tests.repo {repo_url}")
+        shell("curl -o /etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release https://www.redhat.com/security/data/fd431d51.txt")
+
+
+def _remove_client_tools_repo_oracle(shell):
+    """
+    Helper function.
+    Remove the client tools repositories and the redhat-release GPG key
+    on the Oracle Linux systems.
+    Created as a function given we need to be able to call this during
+    the test execution not just the teardown phase.
+    """
+    if "oracle" in SYSTEM_RELEASE_ENV:
+        # Remove the client-tools repofile so those it's not considered during convert2rhel repoquery calls
+        shell("rm -f /etc/yum.repos.d/client-tools-for-tests.repo")
+        # Remove the redhat-release GPG key
+        shell("rm -f /etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release")
+
+
+@pytest.fixture
+def satellite_registration(shell, yum_conf_exclude):
+    """
+    Register the system to the Satellite server
+    """
+    if "oracle" in SYSTEM_RELEASE_ENV:
+        yum_conf_exclude(exclude=["rhn-client*"])
+    _add_client_tools_repo_oracle(shell)
+    # Get the registration command for the respective system
+    # from the conftest function
+    sat_reg_command = satellite_registration_command()
+
+    # Register the system to the Satellite server
+    assert shell(sat_reg_command).returncode == 0
```

Comment:
To make it more reliable, wouldn't be better to check against the sat_reg_command not being None?

```suggestion
    sat_reg_command = satellite_registration_command()
    
    # Make sure it returned some value, otherwise it waill fail. 
    assert sat_reg_command
    
    # Register the system to the Satellite server
    assert shell(sat_reg_command).returncode == 0
```